### PR TITLE
Update smart deletion

### PR DESCRIPTION
**Pull request by hardikdr**
**What this PR does / why we need it**:  Throttled deletion of the failed machines.
Please do not merge, I'm yet to test it appropriately.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: vendor/github.com/gophercloud/gophercloud/go.mod [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module github.com/gophercloud/gophercloud
 
+go 1.15
```
Comment (hardikdr) [inline vendor/github.com/gophercloud/gophercloud/go.mod]: Oops, I dont know where did this come from .. :P

**Comment by zuzzas** [comment]
Unfortunately, all unit tests are now broken.

**Comment by hardikdr** [comment]
Ah I see, I'll take a look, I couldn't find the time yet to test it well. Sorry, too busy last week.

**Comment by zuzzas** [comment]
@hardikdr 
We've clearly got some trouble with this PR (on my side as well). I'd like to direct your attention to tests first and foremost. Let's encode our _expectations_ of what this algorithm should do. If the tests are bad, just say so. I've designed them with the exact requirements that I envision for this algorithm.

**Comment by hardikdr** [comment]
/unhold this can be merged, I guess.

We still need a bit of improvement on top, I'll add the exact expectations in the original issue, and maybe we can then add the tests as well.

**Pull request merged**